api/src/main/java/com/datatorrent/api/StorageAgent.java

```diff
@@ -94,4 +94,24 @@
     public void setApplicationAttributes(AttributeMap map);
   }
 
+  interface AsyncStorageAgent
+  {
+    /**
+     * Make checkpoint for given windowID final. i.e after this method returns,
+     * the checkpoint is accessible for recovery.
+     *
+     * @param operatorId
+     * @param windowId
+     * @throws IOException
+     */
+    void finalize(int operatorId, long windowId) throws IOException;
```

Comment:
Is this a way to write the checkpoint immediately in some scenarios? Why call it finalize? It should probably be something like saveImmediately.